src/histograms/AbstractHistogram.ts

```diff
@@ -263,6 +253,105 @@ export abstract class AbstractHistogram {
     }
   }
 
+  public createXLabelAxis(svgNode: HistogramSVGG, xLabelsAxis, leftOffset: number){
+    return svgNode.append('g')
+        .attr('class', 'histogram__labels-axis')
+        .attr('transform', 'translate(' + (leftOffset - 1) + ',' + this.chartDimensions.height * this.histogramParams.xAxisPosition + ')')
+        .call(xLabelsAxis);
+  }
+
+  public getHorizontalOffset(chartAxes){
+    let h = this.chartDimensions.height;
+    if (isChartAxes(chartAxes)) {
+      if (!this.histogramParams.yAxisFromZero) {
+        const minMax = chartAxes.yDomain.domain();
+        if (minMax[0] >= 0) {
+          h = chartAxes.yDomain(minMax[0]);
+        } else {
+          h = chartAxes.yDomain(minMax[1]);
+        }
+      } else {
+        h = chartAxes.yDomain(0);
+      }
+    }
+    return h;
+  }
+
+  public updateNumberOfLabelDisplayedIfOverlap(chartAxes: ChartAxes | SwimlaneAxes, leftOffset = 0){
+    // Get the offset used when we will draw the labels.
+    const horizontalOffset = this.getHorizontalOffset(chartAxes);
+      let sumWidth = 0;
+      // create virtual nodes. Helps to get label's width.
+      const virtualLabels = this.chartDimensions.svg.append('g');
+      let labels;
+      if(this._previousLabel){
+        labels = this._previousLabel.selectAll('text');
+      } else {
+        labels = this.createXLabelAxis(virtualLabels,chartAxes.xLabelsAxis, leftOffset ).selectAll('text');
+      }
+
+      // check for all labels if there is an overlap.
+      let hasOverlap = false;
+      const nodes = labels.nodes();
+
+      for (let i = 0; i < labels.size(); i++) {
+        const next = i + 1;
+        if(nodes[next]){
+          const currentNodeDimensions = this.getDimension(nodes[i]);
+          const nextNodeDimensions = this.getDimension(nodes[next]);
+          if(!this.isOverlapXAxis(currentNodeDimensions,nextNodeDimensions)) {
+            hasOverlap = true;
+          }
+          sumWidth += currentNodeDimensions.width;
+        }
+    }
+
+    // remove virtual node. If we do not it will be displayed
+    virtualLabels.remove();
+    // calc label mean width once.
+   if(!this._xlabelMeanWidth) {
+      this._xlabelMeanWidth  = Math.round(sumWidth / this.histogramParams.xLabels);
+   }
+
+    if(!hasOverlap) {
+      return;
+    }
+
+    // get the min value between default label size and the max label size allowed. According to the mean width of a label
+    // and the width of the chart
+    const labelCount = min([
+        this.histogramParams.xLabels,
+      Math.floor(this.histogramParams.chartWidth  /  (this._xlabelMeanWidth + horizontalOffset))]
+    );
+
+    // update ticks for label and ticks axis. If we have a lot of label we resize ticks.
+    chartAxes.xLabelsAxis.ticks(labelCount);
```

Comment:
there is an issue with time histograms
when the scale of an axis is time, the `.ticks(firstArg)` method firstArg is not a count but a time interval which leads to incoherence